### PR TITLE
Added user-profile screen

### PR DESCRIPTION
**Pull request by 0804EshitaSharma**
Added  Initial user-profile screen
Fixed Signout bug
Added Password Reset feature
Added feature to Update user info
Screenshot 2023-06-23 at 7 05 38 AM

### CODE REVIEW

File: client/src/components/User/UserProfile.js [hunk 1]
```diff
@@ -1,13 +1,143 @@
...
+import { updateProfile } from "firebase/auth";
+import { updateUserInfo } from "../../redux/userSlice.js";
 
 function UserProfile() {
-  const user = useSelector((state) => state.user);
-  console.log(user);
+  const user = useSelector((state) => state.user.user);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { handleSubmit, reset, register } = useForm({});
```
Comment (ellee311) [inline client/src/components/User/UserProfile.js:16]: is the empty object needed as a parameter for useForm? if not, please remove it, thanks!

File: client/src/components/User/UserProfile.js [hunk 2]
```diff
@@ -1,13 +1,143 @@
...
+  const changePassword = () => {
+    navigate("/forgot-password");
+  };
+  const updateUser = (event) => {
+    console.log(event);
```
Comment (ellee311) [inline client/src/components/User/UserProfile.js:21]: please remove the console.log's if they don't provide a purpose for the users, thank you!

File: client/src/components/User/UserProfile.js [hunk 3]
```diff
@@ -1,13 +1,143 @@
...
+            <img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQx9tjaExsY-srL4VsHNE_OKGVCJ-eIFNBktw&usqp=CAU" />
+          </div>
+          <div className="input_row">
+            <div className="profile_input_field">
+              <CustomFormInput
+                name="Account Holder"
+                id="accountholder"
+                type="text"
+                value={user?.name}
+                label=" Account Holder"
```
Comment (ellee311) [inline client/src/components/User/UserProfile.js:59]: Is the space before the words meant to be there? If so, you can just ignore this comment

File: client/src/components/User/UserProfile.js (hunk 2 above)
Comment (0804EshitaSharma) [reply client/src/components/User/UserProfile.js:24]: done

File: client/src/components/User/UserProfile.js (hunk 3 above)
Comment (0804EshitaSharma) [reply client/src/components/User/UserProfile.js:57]: Yes, it is Label, I believe it looks good with space.

File: client/src/components/User/UserProfile.js (hunk 1 above)
Comment (0804EshitaSharma) [reply client/src/components/User/UserProfile.js:17]: done

**Review by Dan2lle — APPROVED** [review]

**Pull request merged**